libc/src/string/memcmp.cpp

```diff
@@ -16,6 +17,8 @@ namespace LIBC_NAMESPACE_DECL {
 
 LLVM_LIBC_FUNCTION(int, memcmp,
                    (const void *lhs, const void *rhs, size_t count)) {
+  LIBC_CRASH_ON_NULLPTR(lhs);
+  LIBC_CRASH_ON_NULLPTR(rhs);
```

Comment:
For some of these `mem` functions from `string.h`, we might only want to do these checks if the `count` parameter is not zero.  I seem to recall that some code bases can expect to call these with nullptr if the `count` is zero.

https://developers.redhat.com/articles/2024/12/11/making-memcpynull-null-0-well-defined

Comes to mind. It links to https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3322.pdf which has a list of affected functions.  I recommend taking the time to read that blog post and the linked to doc on proposed changes to the C standard.